Tests/SILTests/InstructionTests.swift

```diff
@@ -1,90 +1,113 @@
+    "unreachable",
+    "witness_method $Self, #Comparable.\"<=\"!1 : <Self where Self : Comparable> (Self.Type) -> (Self, Self) -> Bool : $@convention(witness_method: Comparable) <τ_0_0 where τ_0_0 : Comparable> (@in_guaranteed τ_0_0, @in_guaranteed τ_0_0, @thick τ_0_0.Type) -> Bool"
+]
 
-    // In order to declare this as `let instructionDef: String`, we need to write an appropriate init.
-    // In that init, we need to delegate to the superclass init that involves `NSInvocation`.
-    // That doesn't seem possible, so we use this hack.
-    // error: 'NSInvocation' is unavailable in Swift: NSInvocation and related APIs not available.
-    private var instructionDef: String!
+// This definition uses a macOS-only trick to dynamically generate test cases.
+// That not just doesn't work on Linux but also doesn't compile.
+#if os(macOS)
+    public final class InstructionTests: XCTestCase {
+        // In order to declare this as `let instructionDef: String`, we need to write an appropriate init.
+        // In that init, we need to delegate to the superclass init that involves `NSInvocation`.
+        // That doesn't seem possible, so we use this hack.
+        // error: 'NSInvocation' is unavailable in Swift: NSInvocation and related APIs not available.
+        private var instructionDef: String!
 
-    public func testRoundtrip() {
-        do {
-            let p = SILParser(forString: instructionDef)
-            let i = try p.parseInstructionDef()
-            XCTAssertEqual(instructionDef, i.description)
-        } catch {
-            XCTFail(String(describing: error) + "\n" + instructionDef)
+        public func testRoundtrip() {
+            do {
+                let p = SILParser(forString: instructionDef)
+                let i = try p.parseInstructionDef()
+                XCTAssertEqual(instructionDef, i.description)
+            } catch {
+                XCTFail(String(describing: error) + "\n" + instructionDef)
+            }
         }
-    }
 
-    override class var defaultTestSuite: XCTestSuite {
-        let testSuite = XCTestSuite(name: NSStringFromClass(self))
-        for instructionDef in instructionDefs {
-            for invocation in testInvocations {
-                let testCase = InstructionTests(invocation: invocation)
-                testCase.instructionDef = instructionDef
-                testSuite.addTest(testCase)
+        public override class var defaultTestSuite: XCTestSuite {
+            let testSuite = XCTestSuite(name: NSStringFromClass(self))
+            for instructionDef in instructionDefs {
+                for invocation in testInvocations {
+                    let testCase = InstructionTests(invocation: invocation)
+                    testCase.instructionDef = instructionDef
+                    testSuite.addTest(testCase)
+                }
             }
+            return testSuite
+        }
+    }
+#endif
+
+#if os(Linux)
+    public final class InstructionTests: XCTestCase {
```

Comment:
Do we really need both paths? Doesn't this one work on macOS?

Reply:
It doesn't because allTests are only used in LinuxMain.swift.